include/Dialect/LWE/IR/LWEAttributes.td

```diff
@@ -55,10 +67,143 @@ def LWE_BitFieldEncoding: LWE_EncodingAttr<"BitFieldEncoding", "bit_field_encodi
+    `bit_field_encoding` for per-coefficient encodings.
+
+    The presence of this attribute as the `encoding` attribute of a tensor of
+    `poly.poly` indicates that the tensor is an RLWE ciphertext for some RLWE
+    scheme that supports the coefficient encoding.
+
+    Example:
+
+    ```
+    #generator = #poly.polynomial<1 + x**1024>
+    #ring = #poly.ring<cmod=65536, ideal=#generator>
+    #coeff_encoding = #lwe.poly_coefficient_encoding<cleartext_start=15, cleartext_bitwidth=4>
+
+    %poly1 = poly.from_tensor %coeffs1 : tensor<10xi16> -> !poly.poly<#ring>
+    %poly2 = poly.from_tensor %coeffs2 : tensor<10xi16> -> !poly.poly<#ring>
+    %rlwe_ciphertext = tensor.from_elements %poly1, %poly2 : tensor<2x!poly.poly<#ring>, #coeff_encoding>
+    ```
+  }];
+}
+
+def RLWE_PolyEvaluationEncoding
+  : LWE_EncodingAttrWithScalingFactor<"PolyEvaluationEncoding", "poly_evaluation_encoding"> {
+  let summary = "An attribute describing encoded RLWE plaintexts via evaluations at fixed points.";
+  let description = [{
+    A "evaluation encoding" of a list of integers $(v_1, \dots, v_n)$ asserts
+    that $f(\x_1) = v_1, \dots, f(x_n) = v_n$ for some implicit, but fixed and
+    distinct choice of inputs $x_i$. The encoded values are also scaled by a
+    scale factor, having the same semantics as `bit_field_encoding`, but
+    applied entry-wise (to either the coefficient or evaluation representation).
+
+    This attribute can be used in multiple ways:
+
+    - On a `poly.poly`, it asserts that the polynomial has been transformed
+      from an evaluation tensor.
+    - On a tensor of `poly.poly`, it asserts that the tensor is an RLWE
+      ciphertext for some RLWE scheme that supports the evaluation encoding.
+
+    A typical workflow for the BFV/BGV schemes using this encoding would be
+    to apply a INTT operation to the input list of cleartexts to convert from
+    evaluation form to coefficient form, then encrypt the resulting polynomial
+    in coefficient form, then apply NTT back to the evaluation form for faster
+    multiplication of ciphertexts.
+
+    The points chosen are fixed to be the powers of a primitive root of unity
+    in the relevant coefficient domain, which allows one to use NTT/INTT to
+    tansform quickly between the coefficient and evaluation forms.
+
+    Example:
+
+    ```
+    #generator = #poly.polynomial<1 + x**1024>
+    #ring = #poly.ring<cmod=65536, ideal=#generator>
+    #lwe_encoding = #lwe.poly_evaluation_encoding<cleartext_start=30, cleartext_bitwidth=3>
+
+    %evals = arith.constant <[1, 2, 4, 5]> : tensor<4xi16>
+    // Note no `intt` operation exists in poly yet.
+    %poly1 = poly.intt %evals : tensor<4xi16> -> !poly.poly<#ring, #eval_encoding>
+    %poly2 = poly.intt %evals : tensor<4xi16> -> !poly.poly<#ring, #eval_encoding>
+    %rlwe_ciphertext = tensor.from_elements %poly1, %poly2 : tensor<2x!poly.poly<#ring, #eval_encoding>>
+    ```
+  }];
 }
+
+def RLWE_PolyRoundedEmbeddingEncoding
```

Comment:
This name seems a bit unclear, as embedding could mean many things. Maybe use something else? In the literature it is usually called just "CKKS encoding" .

Reply:
I'm trying to give them names that suggest their semantic meaning, rather than tie them to a (not future proof) black-box label like CKKS. If a new scheme not invented by the same authors uses the same encoding, then the name ceases to be sensible. Maybe "InverseCanonicalEmbeddingEncoding" ?